### DISCUSSION TITLE
Specify python version in virtualenv

### DISCUSSION THREAD

**Discussion by tvdboom** [pdm-project/pdm#2775] [Q&A]
Hi,

I'm running `pdm sync -G lint --no-default` to install the packages, which creates a new virtual env with python 3.10.

![image](https://github.com/pdm-project/pdm/assets/32366550/df3c5a59-f0b1-44bf-bc00-328382417dc5)
![image](https://github.com/pdm-project/pdm/assets/32366550/a4903b2e-e103-4b51-a55c-25b44a848724)

How can I tell it to use Python 3.11? I am already using setup-pdm specifying the python version like this:

      - name: Set up pdm
        uses: pdm-project/setup-pdm@v4
        with:
          python-version: "3.11"
          cache: true
Or better yet. how can I tell it to use the current virtual env (which I assume is using the 3.11)?

**Comment by huxuan** [comment]
`python-version: "3.11"` should have set the python version for pdm to use. Do you have a live example which contains the whole workflow logs?

**Comment by frostming** [comment]
On `2.13` you can do `pdm use 3.11`

**Comment by tvdboom** [comment]
Weirdly enough, I am not encountering the issue in the latest builds. But you can find the one that was wrong [here](https://github.com/tvdboom/ATOM/actions/runs/8573779785/job/23499195837).

As you can see, I first set up pdm with Python 3.11
![image](https://github.com/pdm-project/pdm/assets/32366550/61db612c-0e9c-47ed-aecd-152328b0c080)

Then it creates a new `venv` in `Install dependencies`
![image](https://github.com/pdm-project/pdm/assets/32366550/604d6c06-6e53-4415-8914-9f4393239519)

And that `venv` has Python 3.10
![image](https://github.com/pdm-project/pdm/assets/32366550/c92d229c-a171-4f5a-8b8b-14661b14c86a)
